flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointedInputGate.java

```diff
@@ -74,34 +106,34 @@ public CheckpointedInputGate(
 	}
 
 	@Override
-	public Optional<BufferOrEvent> pollNext() throws Exception {
-		while (true) {
```

Comment:
Yes, it's changing semantics (as written in commit message). I have not found a good reason why it's not always exited and it makes things easier especially since this method can now be used to process priority events.